File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_016)_Move_All_ZEROs_to_End.cpp
#include<iostream>
using namespace std;

void moveToEnd(int arr[], int n)
{
    int count = 0;
    for(int i=0; i<n; i++)
    {
        if(arr[i] != 0)
        {
            int temp = arr[i];
            arr[i] = arr[count];
            arr[count] = temp;
            count++;
        }
    }
}

int main()
{
    int arr[] = {0, 8, 5, 0, 0, 10, 0, 20}, n=8;
    cout<<"Before Move Zeros: ";
    for(int i=0; i<n; i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl<<"Size = "<<n<<endl;

    moveToEnd(arr, n);
    cout<<"After Move Zeros: ";
    for(int i=0; i<n; i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl<<"Size = "<<n<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_016)_Move_All_ZEROs_to_End.dart
void moveToEnd(arr) {
  var count = 0;
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] != 0) {
      //swaping the values
      var temp = arr[i];
      arr[i] = arr[count];
      arr[count] = temp;
      count++;
    }
  }
}

void main() {
  var arr = [5, 0, 0, 4, 0, 7, 0, 0, 0, 3];
  print('Before moving zeros: ${arr}');
  moveToEnd(arr);
  print('After moving zeros: ${arr}');
}
